src/request.spec.ts

```diff
@@ -394,6 +395,82 @@ describe('Request', () => {
       });
   });
 
+  it('returns response of a successful GET request, with a proxy url', (done) => {
```

Comment:
What are we testing in this test exactly? Since we're stubbing `request.req`, the test will always pass. Given how the test is named right now, we'd need to actually call an API to see if the request comes through. That might not be desirable, because it would require us to have a proxy running. So a viable alternative could be to see if a proxy is configured on the request object. The same applies to the next 3 tests as well.